Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Generator.cs

```diff
@@ -85,16 +91,38 @@ public void Execute(GeneratorExecutionContext context)
                 }
 
                 var isExecutable = false;
+
+                bool foundFatalError = false;
                 
                 var assemblyAttributes = context.Compilation.Assembly.GetAttributes();
                 
                 // Let's find the AssemblyTitleAttribute
                 var generateMainAttribute = assemblyAttributes
-                    .FirstOrDefault(attr => attr.AttributeClass.Name == nameof(LambdaGenerateMainAttribute));
+                    .FirstOrDefault(attr => attr.AttributeClass.Name == nameof(LambdaGlobalPropertiesAttribute));
+
+                var defaultRuntime = "dotnet6";
 
                 if (generateMainAttribute != null)
                 {
-                    isExecutable = true;
+                    var generateMain = generateMainAttribute.NamedArguments.FirstOrDefault(kvp => kvp.Key == "GenerateMain").Value;
```

Comment:
We are missing some null checks here for both `generateMain` and `runtime` since the `FirstOrDefault` will return null. That also means we are missing some unit test cases since it looks like we are only testing with both values set.